clients/src/test/java/org/apache/kafka/common/metrics/SensorTest.java

```diff
@@ -59,4 +65,22 @@ public void testShouldRecord() {
             0, Sensor.RecordingLevel.DEBUG);
         assertFalse(debugSensor.shouldRecord());
     }
+
+    @Test
+    public void testExpiredSensor() {
```

Comment:
nit: the timeout is a little obscured below. I'd suggest storing the timeout itself in a nicely named variable and using `time.sleep()` instead of using the auto-tick capability of `MockTime`.